src/compiler/scala/tools/nsc/reporters/Filter.scala

```diff
@@ -0,0 +1,54 @@
+/* NSC -- new Scala compiler
+ * Copyright 2002-2017 LAMP/EPFL
+ * @author A. P. Marki
+ */
+package scala.tools.nsc
+package reporters
+
+import reflect.internal.util.Position
+import reflect.internal.settings.MutableSettings
+
+/** A `Reporter` that filters events before delegating.
+ */
+abstract class Filter(delegate: Reporter) extends Reporter {
```

Comment:
Could you instead create a class `DelegatingReporter`, place it next to `Reporter`, and maybe put a comment in `Reporter` somewhere mentioning the existence of the delegate? This would increase the likelyhood of the delegate to stay up to date.